services/121-service/src/payments/fsp-integration/commercial-bank-ethiopia/commercial-bank-ethiopia.api.service.ts

```diff
@@ -0,0 +1,228 @@
+import { Injectable } from '@nestjs/common';
+import { SoapService } from '../../../utils/soap/soap.service';
+import { CommercialBankEthiopiaMockService } from './commercial-bank-ethiopia.mock';
+import { CommercialBankEthiopiaTransferPayload } from './dto/commercial-bank-ethiopia-transfer-payload.dto';
+import { CommercialBankEthiopiaSoapElements } from './enum/commercial-bank-ethiopia.enum';
+
+@Injectable()
+export class CommercialBankEthiopiaApiService {
+  public constructor(
+    private readonly soapService: SoapService,
+    private commercialBankEthiopiaMock: CommercialBankEthiopiaMockService,
+  ) {}
+
+  public async creditTransfer(
+    payment: CommercialBankEthiopiaTransferPayload,
+    credentials: { username: string; password: string },
+  ): Promise<any> {
+    const payload = await this.createCreditTransferPayload(
+      payment,
+      credentials,
+    );
+
+    try {
+      const responseBody = !!process.env.MOCK_COMMERCIAL_BANK_ETHIOPIA
+        ? await this.commercialBankEthiopiaMock.post(payload, payment)
+        : await this.soapService.postCBERequest(
+            payload,
+            `${process.env.COMMERCIAL_BANK_ETHIOPIA_URL}?xsd=4`,
+          );
+
+      if (
+        responseBody.Status &&
+        responseBody.Status.messages &&
+        responseBody.Status.messages.length > 0 &&
+        responseBody.Status.messages[0]._text.includes(
```

Comment:
@Vladimir-Petkov is this check future-proof enough? Isn't there an error code we can check on? (the diagram mentioned something like T24?).

Reply:
@jannisvisser Yes, we have this T24 but this is for all errors that they have and we have to check the messages array, that has 3 different messages. In first one we have something like: "Transaction with number 21321312 is DUPLICATED Transaction!" and we need to check if the string includes this: "DUPLICATED Transaction!"